src/gui/src/label.cpp

```diff
@@ -0,0 +1,242 @@
+{
+  auto label = std::any_cast<Label*>(object);
+  const int dbu_per_uu_ = db_->getChip()->getBlock()->getDbUnitsPerMicron();
+
+  std::vector<Descriptor::EditorOption> anchor_options;
+  for (const auto& anc : {Painter::Anchor::BOTTOM_LEFT,
+                          Painter::Anchor::BOTTOM_RIGHT,
+                          Painter::Anchor::TOP_LEFT,
+                          Painter::Anchor::TOP_RIGHT,
+                          Painter::Anchor::CENTER,
+                          Painter::Anchor::BOTTOM_CENTER,
+                          Painter::Anchor::TOP_CENTER,
+                          Painter::Anchor::LEFT_CENTER,
+                          Painter::Anchor::RIGHT_CENTER}) {
+    anchor_options.push_back({anchorToString(anc), anc});
+  }
+
+  return {{"Name", makeEditor([this, label](std::any value) {
+             auto new_name = std::any_cast<const std::string>(value);
+             if (new_name.empty()) {
+               return false;
+             }
+             for (const auto& check_label : labels_) {
+               if (check_label->getName() == new_name) {
+                 return false;
+               }
+             }
+             label->setName(new_name);
+             return true;
+           })},
+          {"Text", makeEditor([label](std::any value) {
+             auto new_text = std::any_cast<const std::string>(value);
+             if (new_text.empty()) {
+               return false;
+             }
+             label->setText(new_text);
+             return true;
+           })},
+          {"Size", makeEditor([label](std::any value) {
+             try {
+               auto new_size = static_cast<int>(std::any_cast<double>(value));
+               if (new_size <= 0) {
+                 return false;
+               }
+               label->setSize(new_size);
+             } catch (const std::bad_any_cast&) {
+               return false;
+             }
+             return true;
+           })},
+          {"Anchor",
+           makeEditor(
+               [label](std::any value) {
+                 auto anchor = std::any_cast<Painter::Anchor>(value);
+                 label->setAnchor(anchor);
+                 return true;
+               },
+               anchor_options)},
+          {"Color", makeEditor([this, label](std::any value) {
+             label->setColor(Painter::stringToColor(
+                 std::any_cast<const std::string>(value), logger_));
+             return true;
+           })},
+          {"x", makeEditor([label, dbu_per_uu_](const std::any& value) {
+             return LabelDescriptor::editPoint(
+                 value, dbu_per_uu_, label->getPt(), true);
+           })},
+          {"y", makeEditor([label, dbu_per_uu_](const std::any& value) {
+             return LabelDescriptor::editPoint(
+                 value, dbu_per_uu_, label->getPt(), false);
+           })}};
+}
+
+bool LabelDescriptor::editPoint(std::any value,
```

Comment:
I can't edit the x/y values and get them to stick.  When I hit enter the just return to their previous values.

Reply:
This seems to be true for the rulers as well